motionplan/rrt.go

```diff
@@ -66,16 +64,55 @@ func (plan *rrtPlanReturn) err() error {
 type rrtMaps struct {
 	startMap rrtMap
 	goalMap  rrtMap
+	optNode  *costNode // The highest quality IK solution
 }
 
-func initRRTMaps() *rrtMaps {
-	return &rrtMaps{
-		startMap: map[node]node{},
-		goalMap:  map[node]node{},
+func initRRTMaps(ctx context.Context, mp motionPlanner, goal spatialmath.Pose, seed []referenceframe.Input) *rrtPlanReturn {
+	rrt := &rrtPlanReturn{
+		rm: &rrtMaps{
+			startMap: map[node]node{},
+			goalMap:  map[node]node{},
+		},
 	}
+	seedNode := newCostNode(seed, 0)
+	rrt.rm.startMap[seedNode] = nil
+
+	// get many potential end goals from IK solver
+	solutions, err := mp.getSolutions(ctx, goal, seed)
```

Comment:
Have we ever seen a situation where an IK solution could have multiple "direct" solutions (say we commanded a motion that could have a -Pi/2 or Pi/2 rotation of the flange joint on a UR5) but only one of the two would be viable from `checkPath`'s perspective?

Reply:
OK, now that I know this occurs often enough to care about it, are there situations where the ordering of IK solutions causes an early bailout because of some asymmetry in the planning environment? Say the solutions are ordered with Pi/2 first, then -Pi/2, but only -Pi/2 is viable because we're up against a wall or an interaction space. Is that enough of a headache for us to consider alternate logic for path viability checking?

Reply:
Not sure I'm fully following this.

Since we check path validity, and try to interpolate to many solutions rather than just one, it should pick the first valid one and correctly find -Pi/2.

It should never cause an "early bailout" except in the case where it finds a valid solution. In which case that's exactly what we want.

Reply:
Yeah, I guess the comment on L93 ("check until out of bounds") made me think path viability, not path cost. The logic does what you say.